### PR TITLE
Lowercase enum values to avoid hypen in k8s flavor acronym

### LINKED ISSUE #33004: Kubernetes Dev Services: k3s flavour name is not properly documented
### Description

The k3s flavour needs to be passed as `k3-s` due to how SmallRye enumeration names conversions. That name is rather awkward and is not properly documented. Ideally, it should be named `k3s` but if it's not possible, the proper name should be documented accordingly.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by scrocquesel**
Fix https://github.com/quarkusio/quarkus/issues/33004

### CODE REVIEW

**Review by metacosm — APPROVED** [review]

**Review by Sgitario — APPROVED** [review]

File: extensions/kubernetes-client/runtime-internal/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesDevServicesBuildTimeConfig.java [hunk 1]
```diff
@@ -29,7 +29,7 @@ public class KubernetesDevServicesBuildTimeConfig {
     /**
      * The flavor to use (kind, k3s or api-only). Default to api-only.
      */
-    @ConfigItem(defaultValue = "API_ONLY")
+    @ConfigItem(defaultValue = "api-only")
```
Comment (Sgitario) [inline extensions/kubernetes-client/runtime-internal/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesDevServicesBuildTimeConfig.java:32]: Should not be `api_only`?

Comment (gsmet) [reply extensions/kubernetes-client/runtime-internal/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesDevServicesBuildTimeConfig.java:32]: No. We replace `_` with `-` when dealing with enum default values in Quarkus.

Comment (Sgitario) [reply extensions/kubernetes-client/runtime-internal/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesDevServicesBuildTimeConfig.java:32]: I've just tried that both values `api-only` and `api_only` are indeed working fine. 
Though, I would prefer using `api_only` which is the value present in the Flavor enum. This way we don't need to know this special handling made when using enum default values.

**Review by Sgitario — APPROVED** [review]
Though, I would prefer using `api_only` in the `@ConfigItem` which is the value present in the Flavor enum. This way we don't need to know this special handling made when using enum default values.

File: extensions/kubernetes-client/runtime-internal/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesDevServicesBuildTimeConfig.java (hunk 1 above)
Comment (gsmet) [reply extensions/kubernetes-client/runtime-internal/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesDevServicesBuildTimeConfig.java:32]: We are using dashed values consistently in all Quarkus so let's keep it this way.

We decided to go with dashes for all default values.

**Pull request merged**